BlockEntity/BECharcoalPit.cs

```diff
@@ -255,35 +252,70 @@ BlockPos FindHoleInPit()
 
                 foreach (BlockFacing facing in BlockFacing.ALLFACES)
                 {
-                    BlockPos npos = bpos.AddCopy(facing);
+                    facing.IterateThruFacingOffsets(npos);  // This must be the first command in the loop, to ensure all facings will be properly looped through regardless of any 'continue;' statements
                     IWorldChunk chunk = Api.World.BlockAccessor.GetChunkAtBlockPos(npos);
                     if (chunk == null) return null;
 
                     Block nBlock = chunk.GetLocalBlockAtBlockPos(Api.World, npos);
 
-                    bool solid = nBlock.SideSolid[facing.Opposite.Index] || (nBlock is BlockMicroBlock && (chunk.GetLocalBlockEntityAtBlockPos(npos) as BlockEntityMicroBlock).sideAlmostSolid[facing.Opposite.Index]);
+                    bool solid = nBlock.GetLiquidBarrierHeightOnSide(facing.Opposite, npos) == 1 || nBlock.GetLiquidBarrierHeightOnSide(facing, bpos) == 1;
                     bool isFirewoodpile = BlockFirepit.IsFirewoodPile(Api.World, npos);
 
-                    if (!solid && !isFirewoodpile && nBlock.BlockId != charcoalPitBlockId)
+                    if (!isFirewoodpile && nBlock.BlockId != charcoalPitBlockId)
                     {
-                        return bpos;
+                        if (IsCombustible(npos)) holes.Add(npos.Copy());
+                        else if (!solid) holes.Add(bpos.Copy());

```

Comment:
This change allows us to quantify every position where some type of flammable block is exposed to the outside of the charcoal pit, including those making up the walls, rather than just the firewood.